src/utils/activity/ExecuteActivityOutputStepBase.ts

```diff
@@ -33,15 +33,37 @@ export abstract class ExecuteActivityOutputStepBase<T extends IActionContext & E
     protected fail: ExecuteActivityOutput;
 
     public async execute(context: T, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
-        this.success = this.initSuccessOutput(context);
-        this.fail = this.initFailOutput(context);
+        this.success = this.createSuccessOutput(context);
```

Comment:
Overall, this feels much cleaner to me. Couple of things you can change:

- You can do `this.displayOutput` at the end so you don't have to call it twice like this. I'd do something like this;
```
       let output;
       try {
            await this.executeCore(context, progress);
            output = this.createSuccessOutput(context);
        } catch (e) {
            output = this.createFailOutput(context);
            if (!this.options.shouldSwallowError) {
                throw e;
            } 
        } 
       
        this.displayOutput(context, output)    
```

- You could also get rid of success and fail as properties on the interface. I personally think this is a bit cleaner. I don't think the finally clause will run

Reply:
There's some rare cases where I might want to add onto the output log messages inside of `executeCore`, so I think it might be useful to have the `success` and `fail` outputs initialized and available ahead of time so they can be accessed within that method.